### PR TITLE
Feature/448681 run calculation name validation

### PR DESCRIPTION
**Pull request by Mazar-Shaik**

### CODE REVIEW

**Review by johnbenjamin-defra — APPROVED** [review]

**Review by udaydefra — DISMISSED** [review]

**Review by sunitaguptaEviden — DISMISSED** [review]

File: src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs [hunk 1]
```diff
@@ -33,13 +37,36 @@ public ViewResult Confirmation()
             return this.View(ViewNames.CalculationRunConfirmation);
         }
 
-        private static ErrorViewModel CreateErrorViewModel()
+        private static ValidationResult ValidateCalculationRunName(string calculationName)
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs:40]: Can you use FluentValidator? This is a standard across the Defra. I can help you with setting up if you need. Thanks

Comment (Mazar-Shaik) [reply src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs]: @udaydefra, I have updated to FluentValidator.

File: src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs [hunk 2]
```diff
@@ -14,16 +15,20 @@ public IActionResult Index()
...
         [HttpPost]
-        public IActionResult RunCalculator(string calculationName)
+        public IActionResult RunCalculator(InitiateCalculatorRunModel calculationRunName)
         {
-            if (string.IsNullOrWhiteSpace(calculationName))
+            if (!this.ModelState.IsValid)
             {
-                this.ViewBag.Errors = CreateErrorViewModel();
+                var errorMessages = this.ModelState.Values.SelectMany(x => x.Errors).Select(e => e.ErrorMessage);
+                this.ViewBag.Errors = CreateErrorViewModel(errorMessages.First());
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs:23]: You are only showing first message. Should you change it display more than one error message

File: src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs (hunk 1 above)
Comment (udaydefra) [reply src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs]: Looks good, Thanks

File: src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs (hunk 2 above)
Comment (Mazar-Shaik) [reply src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs:23]: Why do we need show more than one message here? As per my validation I have to show separately for each. We can have quick call to discuss on this.

Comment (udaydefra) [reply src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs:23]: _Uday Denduluri (ext) I had a call with Uma, We need to validate on not more than 100 characters first. If it's more than 100 and having some special characters in it then it we shouldn't show 2 error messages at a time (that's what mention in User Story). We only show More than 100 error message. Once user fixed and it's under 100 then will check Alpha Numeric. Hope this Answer your question._

Got this confirmed that we only showing one error message at a time. Thanks

**Review by udaydefra — APPROVED** [review]

**Review by johnbenjamin-defra — DISMISSED** [review]

File: src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs [hunk 3]
```diff
@@ -14,16 +15,20 @@ public IActionResult Index()
...
                 return this.View(CalculationRunNameIndexView);
             }
 
-            this.ViewBag.CalculationName = calculationName;
-            this.HttpContext.Session.SetString(SessionConstants.CalculationName, (string)this.ViewBag.CalculationName);
+            this.ViewBag.CalculationName = calculationRunName?.CalculationName;
+            if (!string.IsNullOrWhiteSpace(this.ViewBag.CalculationName))
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs:28]: `this.ViewBag.CalculationName` can be replaced with `calculationRunName.calculationRunName`
Then you do not need to check for `if (!string.IsNullOrWhiteSpace(this.ViewBag.CalculationName))`

File: src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs [hunk 4]
```diff
@@ -14,16 +15,20 @@ public IActionResult Index()
         }
 
         [HttpPost]
-        public IActionResult RunCalculator(string calculationName)
+        public IActionResult RunCalculator(InitiateCalculatorRunModel calculationRunName)
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs:18]: please rename this `calculationRunName `to `calculationRunModel `

File: src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs (hunk 3 above)
Comment (Mazar-Shaik) [reply src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs]: Updated

**Review by udaydefra — APPROVED** [review]

**Review by johnbenjamin-defra — APPROVED** [review]

**Pull request merged**